frontend/src/metabase-lib/lib/queries/NativeQuery.js

```diff
@@ -44,17 +44,19 @@ export const NATIVE_QUERY_TEMPLATE: NativeDatasetQuery = {
   },
 };
 
+// This regex needs to match logic in replaceCardId and _getUpdatedTemplateTags.
+const CARD_TAG_REGEX = /^#\s*([0-9]*)\s*$/;
```

Comment:
I think this should start with`/^\s*` to allow `{ #123}` (in addition to `{# 123}` and  `{#123 }`)

Reply:
It does allow for that. This regex is used after [parsing the tag name](https://github.com/metabase/metabase/pull/11835/files#diff-98acf6549faf2b8c61213474d5146919R326) out of the query. Any leading space within the braces isn't captured.

Reply:
That's also true for trailing space, so I guess I can drop that in this regex.